apps/condo/domains/news/tasks/notifyResidentsAboutNewsItem.js

```diff
@@ -90,31 +89,41 @@ async function sendNotifications (context, newsItem, taskId) {
         ],
     }), {})
 
-    const { keystone: contextMessage } = await getSchemaCtx('Message')
+    const { keystone: contextMessage } = getSchemaCtx('Message')
     for (const resident of residentsData) {
-        await sendMessage(contextMessage, {
-            ...DV_SENDER,
-            to: { user: { id: resident.user.id } },
-            lang: conf.DEFAULT_LOCALE,
-            type: defineMessageType(newsItem),
-            meta: {
-                dv: 1,
-                title: truncate(newsItem.title, { length: TITLE_MAX_LEN, separator: ' ', omission: '...' }),
-                body: truncate(newsItem.body, { length: BODY_MAX_LEN, separator: ' ', omission: '...' }),
-                data: {
-                    newsItemId: newsItem.id,
-                    organizationId: newsItem.organization.id,
-                    userId: resident.user.id,
-                    residentId: resident.id,
-                    userRelatedResidentsIds: get(residentsIdsByUser, resident.user.id, []).join(','),
-                    url: `${conf.SERVER_URL}/newsItem/${newsItem.id}`,
-                    validBefore: get(newsItem, 'validBefore', null),
-                    // The first truthy value will be returned, or null if no values are found.
-                    dateCreated: ['sendAt', 'publishedAt', 'updatedAt', 'createdAt'].reduce((result, field) => (result || get(newsItem, field)), null),
+        try {
```

Comment:
Will `Promise.all` decrease sending time?
I mean something like this
```js
Promise.all(residentsData.map((resident) => new Promise((resolve) => {
    sendMessage(...).then(() => resolve({ err: null })).catch((err) => resolve({ err: err.message }))
})))
```

Reply:
Please don't send 100500 simultaneous requests :)

Reply:
Each message will still be delivered separately one after another (in the worker, through the "deliverMessage" task)